skbio/tree/tests/test_tree.py

```diff
@@ -633,6 +633,18 @@ def test_set_max_distance(self):
         self.assertEqual(tip_a[0] + tip_b[0], 1.6)
         self.assertEqual(sorted([tip_a[1].name, tip_b[1].name]), ['b', 'e'])
 
+    def test_set_max_distance_tie_bug(self):
+        """Corresponds to #1077"""
+        s = StringIO("((a:1,b:1)c:2,(d:3,e:4)f:5)root;")
+        t = TreeNode.read(s)
+
+        # the above tree would trigger an exception in max. The central issue
+        # was that the data being passed to max were a tuple of tuple:
+        # ((left_d, left_n), (right_d, right_n))
+        # the call to max would break in this scenario as it would fall onto
+        # idx 1 of each tuple to assess the "max".
+        t._set_max_distance()
```

Comment:
It'd also be great to add a separate test for `root_at_midpoint` with a tree that has ties, in case `root_at_midpoint` no longer uses `_set_max_distance` internally in the future.